### PR TITLE
Adding new navigation components and application layout

### LINKED ISSUE #489: As a developer, I want to see navigation markup in the pattern library so that I can use this markup in building out pages.
- [x] Navigation components in the application layout
- [x] Working examples of components with markup in pattern library

---

- [x] Masthead
- [x] Masthead navigation
- [x] Sub-navigation
- [x] Footer navigation
- [x] Filter

### PR DESCRIPTION
**Pull request by heyitsolivia**
Closes #489

### CODE REVIEW

**Comment by heyitsolivia** [comment]
@Cspeisman Mind reviewing my sass and js if you have time?

**Comment by handlers** [comment]
No worries on the build breaking in this PR; the .travis.yml in the file tree is not up-to-date, so it can't run tests.

**Comment by Cspeisman** [comment]
I figured @handlers - will look over #500 and this pull request first thing today cc: @heyitsolivia

File: app/assets/stylesheets/shared/_navigations.scss [hunk 1]
```diff
@@ -1,71 +1,592 @@
...
+        background: rgba(0,0,0,0.2);
+        content: '';
+        height: 0;
+        opacity: 0;
+        position: absolute;
+        right: 0;
+        top: 0;
+        transition: opacity 0.5s, width 0.1s 0.5s, height 0.1s 0.5s;
+        width: 0;
+
```
Comment (Cspeisman) [inline app/assets/stylesheets/shared/_navigations.scss]: @heyitsolivia there is a bourbon mixin to handle triangles http://bourbon.io/docs/#triangle 
Not saying that we have to use.  Just pointing it out

**Comment by Cspeisman** [comment]
dis looks good!! merging

**Pull request merged**